### PR TITLE
Store the metedata blob before hitting the assert failure

### PR DESCRIPTION
**Pull request by olkononenko**

### CODE REVIEW

File: Public/Src/Engine/Scheduler/PipExecutor.cs [hunk 1]
```diff
@@ -2504,6 +2504,19 @@ private static int GetCacheLevelForLocality(PublishedEntryRefLocality locality)
                     I($"File name should only differ by casing. File artifact's full path: '{fileArtifactPathString}'; file artifact's file name: '{fileArtifactFileName.ToString(stringTable)}'; materialization info file name: '{fileMaterializationFileNameString}'."));                
                 stringBuilder.AppendLine(I($"[{process.FormattedSemiStableHash}] Weak FP: '{runnableFromCacheCheckResult.WeakFingerprint.ToString()}', Strong FP: '{cacheHitData.StrongFingerprint.ToString()}', Metadata Hash: '{cacheHitData.MetadataHash.ToString()}'"));
 
+                // get the bytes of a metadata blob
```
Comment (mpysson) [inline Public/Src/Engine/Scheduler/PipExecutor.cs]: >  [](start = 0, length = 1)

Please log an error here before performing this collection for good measure

File: Public/Src/Engine/Scheduler/PipExecutor.cs [hunk 2]
```diff
@@ -2504,6 +2504,22 @@ private static int GetCacheLevelForLocality(PublishedEntryRefLocality locality)
...
+
+                // get the bytes of a metadata blob
+                var possibleMetadataBytes = environment.State.Cache.ArtifactContentCache.TryLoadContent(cacheHitData.MetadataHash).GetAwaiter().GetResult();
+                if (!possibleMetadataBytes.Succeeded)
+                {
+                    stringBuilder.AppendLine("Failed to obtain the metadata blob.");
+                }
+                else
+                {
+                    var blobPath = Path.Combine(environment.Configuration.Logging.LogsDirectory.ToString(environment.Context.PathTable), I($"metadata_{cacheHitData.MetadataHash.ToHex()}.blob"));
```
Comment (dannyvv) [inline Public/Src/Engine/Scheduler/PipExecutor.cs]: >LogsDirectory [](start = 82, length = 13)

Should we place these files in a subfolder?

File: Public/Src/Engine/Scheduler/Caching/PipTwoPhaseCache.cs [hunk 3]
```diff
@@ -35,7 +35,10 @@ public class PipTwoPhaseCache
         /// <summary>
         /// The artifact content cache for retrieving/storing content by hash
         /// </summary>
-        protected readonly IArtifactContentCache ArtifactContentCache;
```
Comment (ljcollins25) [inline Public/Src/Engine/Scheduler/Caching/PipTwoPhaseCache.cs]: >protected [](start = 8, length = 9)

Doesn't look like you are using this anymore.

**Comment by olkononenko** [comment]
No need for this logging anymore - closing the PR.

**Pull request closed without merging**